### PR TITLE
[T5A1][W13-B2]YongLinHan

### PR DESCRIPTION
**Pull request by ylhlh**

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -74,12 +78,51 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+#### Use case: Edit tag
+
+**MSS**
+
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to edit tag of a specific person in the list
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: there should be a round of confirmation that address book should seek and user respond.

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -74,12 +78,51 @@ Use case ends.
...
+> 3c1. AddressBook shows an error message <br>
+  Use case ends
+  
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should have minimal startup and processing times.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: what is minimal?
- in general try to be as precise and objective as possible.
- words like nice, easy, more, less etc can be avoided and use precise values (where possible)

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -74,12 +78,51 @@ Use case ends.
...
+6. Should be able to recover from deletion of storage file while program is running.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: isn't this a functionality (one that is already present in the application) ?

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -74,12 +78,51 @@ Use case ends.
...
+7. Should be easy to learn and use.
+8. Should ensure the privacy and security of user data.
+9. Should be reliable.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: how do you define reliability?

File: doc/DeveloperGuide.md [hunk 5]
```diff
@@ -74,12 +78,51 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+#### Use case: Edit tag
+
+**MSS**
+
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: the activity was updated to rename a tag such that all people with the given tag will have the tag updated. please check the description again

**Comment by okkhoy** [comment]
@ylhlh 
Some comments added. Please ack & close the PR after reading comments.

File: doc/DeveloperGuide.md (hunk 4 above)
Comment (ylhlh) [inline doc/DeveloperGuide.md]: Should not damage/lose any user data.

**Pull request closed without merging**